### ISSUE TITLE
Reconnect on connection lost

### ISSUE THREAD

**Issue by h0tbird** [docker/example-voting-app#41]
The `results` app does not handle postgres disconnections.
The `worker` app does not handle redis disconnections.

This is my ugly workaround for the `results` app in `server.js`:

``` node
connect();

function connect() {
  async.retry(
    {times: 1000, interval: 1000},
    function(callback) {
      pg.connect('postgres://postgres@marathon-lb:10014/postgres', function(err, client, done) {
        if (err) {
          console.error("Waiting for db");
        }   
        callback(err, client);
      }); 
    },  
    function(err, client) {
      if (err) {
        return console.err("Giving up");
      } 
      console.log("Connected to db");
      getVotes(client);
    }
  );
}

function getVotes(client) {
  client.query('SELECT vote, COUNT(id) AS count FROM votes GROUP BY vote', [], function(err, result) {
    if (err) {
      console.error("Error performing query: " + err);
      connect();
    } else {
      var votes = collectVotesFromResult(result);
      io.sockets.emit("scores", JSON.stringify(votes));
    }

    setTimeout(function() {getVotes(client) }, 1000);
  });
}
```

**Comment by ManoMarks** [comment]
@spara can you look at fixing this one?

**Comment by spara** [comment]
yes

**Comment by ManoMarks** [comment]
Fixed in #54
